common/functions/src/scalars/uuids/uuid.rs

```diff
@@ -0,0 +1,27 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crate::scalars::function_factory::FunctionFactory;
+use crate::scalars::GenerateUUIDv4Function;
+use crate::scalars::ZeroUUIDFunction;
+
+#[derive(Clone)]
+pub struct UUIDFunction;
+
+impl UUIDFunction {
+    pub fn register(factory: &mut FunctionFactory) {
+        factory.register("generateUUIDv4", GenerateUUIDv4Function::desc());
```

Comment:
Better to merge GenerateUUIDv4Function and ZeroUUIDFunction to be `UUIDFunction<T>` ?


```
type GenerateUUIDv4Function = UUIDFunction<UUIDv4>;

struct UUIDv4;

impl EvalUUID for UUIDv4  { 
 #[inline]
    pub fn eval() -> {
       ...
    }
}
```